# Principal Component Analysis Assignment
---

## Group Members
| Name | Index |
|------|-----|
| Lassey Prince Kwesi | 35***** |
| Abubakar Rashida | 35***** |
| Boateng Richard Osei | 35***** |

---

## Use the PCA approach to determine how temperature influences humidity in the given table below.

| X = Temperature | Y = Humidity |
|-----------------|--------------|
| 40              | 90           |
| 40              | 90           |
| 40              | 90           |
| 30              | 90           |
| 15              | 70           |
| 15              | 70           |
| 15              | 70           |
| 30              | 90           |
| 15              | 70           |
| 30              | 70           |
| 30              | 70           |
| 30              | 90           |
| 40              | 70           |
| 30              | 90           |



## Importing Library

In [1]:
# We are using just numpy for this numerical analysis
import numpy as np

## Data Preparation

In [2]:
# Data from the table
temperature = np.array([40, 40, 40, 30, 15, 15, 15, 30, 15, 30, 30, 30, 40, 30])
humidity = np.array([90, 90, 90, 90, 70, 70, 70, 90, 70, 70, 70, 90, 70, 90])

In [17]:
# Combine into a 2D array (14 samples, 2 features)
data = np.column_stack((temperature, humidity))

## Calculating for Mean

In [18]:
mean_vec = np.mean(data, axis=0)
print(f"   Mean Temperature : {mean_vec[0]:.2f}")
print(f"   Mean Humidity    : {mean_vec[1]:.2f}\n")

   Mean Temperature : 28.57
   Mean Humidity    : 80.00



## Calculting Mean Deviation (Anomalies)

In [21]:
mean_dev = data - mean_vec   
print("   Mean deviations:")
print(np.round(mean_dev, 2))
print()

   Mean deviations:
[[ 11.43  10.  ]
 [ 11.43  10.  ]
 [ 11.43  10.  ]
 [  1.43  10.  ]
 [-13.57 -10.  ]
 [-13.57 -10.  ]
 [-13.57 -10.  ]
 [  1.43  10.  ]
 [-13.57 -10.  ]
 [  1.43 -10.  ]
 [  1.43 -10.  ]
 [  1.43  10.  ]
 [ 11.43 -10.  ]
 [  1.43  10.  ]]



## Finding the Variance

In [22]:
var = np.var(data, axis=0, ddof=0)
print(f"   Var(Temperature) : {var[0]:.2f}")
print(f"   Var(Humidity)    : {var[1]:.2f}\n")

   Var(Temperature) : 90.82
   Var(Humidity)    : 100.00



## Calculating the Standard Deviation

In [23]:
std = np.std(data, axis=0, ddof=0)
print(f"   Std(Temperature) : {std[0]:.2f}")
print(f"   Std(Humidity)    : {std[1]:.2f}\n")

   Std(Temperature) : 9.53
   Std(Humidity)    : 10.00



## Finding the Covariance

In [24]:
cov_temp_hum = np.cov(temperature, humidity, ddof=0)[0,1]

print(f"   Covariance(Temp, Hum): {cov_temp_hum:.2f}\n")

   Covariance(Temp, Hum): 57.14



## The Covariance Matrix

In [25]:
cov_matrix = np.cov(data.T, ddof=0)

print(np.round(cov_matrix, decimals=2))
print()

[[ 90.82  57.14]
 [ 57.14 100.  ]]



## Finding the Eigenvalues

In [27]:
eigenvalues, eigenvectors = np.linalg.eig(cov_matrix)

print("   Eigenvalues:", np.round(eigenvalues, decimals=4))



   Eigenvalues: [ 38.0811 152.7352]


## Finding the Eigenvectors

In [28]:
print("   Eigenvectors:")

print(np.round(eigenvectors, decimals=4))
print()

   Eigenvectors:
[[-0.7349 -0.6782]
 [ 0.6782 -0.7349]]



## Normalizing the Eigenvectors

In [29]:
norms = np.linalg.norm(eigenvectors, axis=0)           # Get the magnitude of each eigenvector
normalized_eigenvectors = eigenvectors / norms         # Divide each vector by its magnitude

In [30]:
print("   Normalized Eigenvectors:")
print(np.round(normalized_eigenvectors, decimals=4))

   Normalized Eigenvectors:
[[-0.7349 -0.6782]
 [ 0.6782 -0.7349]]


## Principal Components

In [33]:
principal_components = mean_dev @ normalized_eigenvectors

print("\n   (PC1, PC2):")
print(np.round(principal_components, decimals=2))


   (PC1, PC2):
[[ -1.62 -15.1 ]
 [ -1.62 -15.1 ]
 [ -1.62 -15.1 ]
 [  5.73  -8.32]
 [  3.19  16.55]
 [  3.19  16.55]
 [  3.19  16.55]
 [  5.73  -8.32]
 [  3.19  16.55]
 [ -7.83   6.38]
 [ -7.83   6.38]
 [  5.73  -8.32]
 [-15.18  -0.4 ]
 [  5.73  -8.32]]


## Explanation


#### Key Observations:

- The data points naturally form **distinct clusters** in the new principal component space.
- Points with **identical original values** (multiple `(40, 90)` entries) have the **same PC scores** (`-1.62, -15.1`), which is expected and correct.
- There is a clear separation between:
  - High temperature + high humidity points
  - Low temperature + low humidity points

#### Interpretation:

- **PC1** (first column) primarily captures the **temperature variation**. Higher PC1 scores generally correspond to higher temperatures.
- **PC2** (second column) captures the remaining variation, mostly related to **humidity differences**.
- The strong clustering shows that **temperature and humidity are highly correlated** in this dataset; as temperature increases, humidity tends to increase as well.

#### Conclusion

This PCA analysis confirms that **temperature has a strong influence on humidity**. 

By transforming the data into principal components, we saw the main patterns more clearly. The first principal component (PC1) explains the majority of the variance, meaning most of the information in the dataset can be represented along a single dominant direction.

This supports the idea that **temperature is a major driving factor** of humidity levels in the given data.